### DISCUSSION TITLE
How can I prevent SegmentedControl to "slide" on show?

### DISCUSSION THREAD

**Discussion by stefanoborini** [mantinedev/mantine#1239] [Q&A] [answered]
I have a horizontal SegmentedControl that is conditionally rendered (think of a tab page) depending on a specific condition.

As you know, the SegmentedControl has a slide transition of the selection cursor when you click on one of the entries. The problem is that when the SegmentedControl is shown due to conditional rendering it slides from the leftmost value to the currently selected one (set via props). This is a bit nasty.

This is an example of the code:

```
type Props = {
  section_id: string;
  param: any;
  value: string;
  onChange: (section_id: string, param_id: string, v: any) => void;
}

export function ParameterControl(props: Props) {
  let control: any = null
  const param = props.param
  const changeCallback = (v: any) => props.onChange(props.section_id, param.id, v)
  if (param.type === "segmented") {
    const data = param.options.map((x: any, idx: number) => ({
          value: idx.toString(),
          label: x.label
        }))
    control = <SegmentedControl data={data} value={props.value} onChange={changeCallback}/>
  } else {
    control = null
  }
  if (control) {
    return (
      <Paper shadow="xs" p="md">
        <Grid>
        <Grid.Col span={4}><Text>{param.name}</Text></Grid.Col>
        <Grid.Col span={8}>
          {control}
        </Grid.Col>
        <Grid.Col span={8} offset={4} sx={(theme) => ({
          backgroundColor: theme.colors.blue[0],
          border: "solid",
          borderRadius: "5px",
          borderWidth: "1px",
          borderColor: theme.colors.blue[2]
        })}><Text size="xs">{param.options[props.value].description}</Text></Grid.Col>
        </Grid>
      </Paper>
    )
  }

  return null;
}

```
ParameterControl is conditionally rendered from the outside, hence the mechanic of "can appear, can disappear" that triggers the behavior.

Is it possible to have the SegmentedControl re-render without going from the leftmost entry to the current value one?

**Comment by rtivital** [comment, marked as answer]
This is an issue with SegmentedControl that should be resolved on our side (PR welcome). I can only recommend setting `transitionDuration` to 0 to avoid that.

**Reply by rtivital** [reply]
@achmurali would you like to work on this issue?

**Reply by stefanoborini** [reply]
Thanks!

**Reply by achmurali** [reply]
> @achmurali would you like to work on this issue?

Yeah sure!!

**Comment by naimulcsx** [comment]
I am facing the issue on 6.0.13. Is this resolved? or I can create a PR for this.
